buildkite/bazelci.py

```diff
@@ -427,7 +427,7 @@ def print_expanded_group(name):
 
 def execute_commands(config, platform, git_repository, git_commit, git_repo_location,
                      use_bazel_at_commit, use_but, save_but, build_only, test_only,
-                     monitor_flaky_tests):
+                     monitor_flaky_tests, incompatible_flags=[]):
```

Comment:
Don't use mutable data structures as default args. You can either use incompatible_flags=None in the args and "... + (incompatible_flags or []) + ..." in the body,
or incompatible_flags=() in the signature and then "... + list(incompatible_flags) + ..." in the body.

The second approach has the benefit that the "+=" in line 983 will still work.

It's not that bad here since you never mutate the list, but in general I would avoid it since it can lead to bugs down the road.

Reply:
Thanks for the advice!